src/jit/codegencommon.cpp

```diff
@@ -7298,13 +7298,19 @@ void CodeGen::genProfilingEnterCallback(regNumber initReg, bool* pInitRegZeroed)
         return;
     }
 
-#if defined(_TARGET_AMD64_) && !defined(UNIX_AMD64_ABI) // No profiling for System V systems yet.
+#if defined(_TARGET_AMD64_)
     unsigned   varNum;
     LclVarDsc* varDsc;
 
+#if !defined(UNIX_AMD64_ABI)
     // Since the method needs to make a profiler callback, it should have out-going arg space allocated.
     noway_assert(compiler->lvaOutgoingArgSpaceVar != BAD_VAR_NUM);
     noway_assert(compiler->lvaOutgoingArgSpaceSize >= (4 * REGSIZE_BYTES));
+#endif
+
+#if defined(UNIX_AMD64_ABI)
+    regSet.AddMaskVars(REG_R10 | REG_R11 | REG_R12 | REG_R13);
```

Comment:
I don't think this is sufficient to have R10-R13 saved in prolog.  
Note then genProfilingEnterCallback() is called after OS prolog and zero init  of frame.  By the time we are generating Enter callback, we have already generated the method prolog that has saved all required callee saved regs.

---
In reply to: [88864538](https://github.com/dotnet/coreclr/pull/7719#discussion_r88864538) [](ancestors = 88864538)